### ISSUE TITLE
ResteasyReactiveRequestContext.setRequestUri not supporting query parameters

### ISSUE THREAD

**Issue by spike83** [quarkusio/quarkus#28555]
Migrating a project form resteasy to resteasy-reactive I found that `ResteasyReactiveRequestContext.setRequestUri` is not applying the query parameters. 

https://github.com/quarkusio/quarkus/blob/5ff6805119c6b63a21996b3b7b58e6f181869f49/independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/ResteasyReactiveRequestContext.java#L469-L478

We have a `ContainerRequestFilter` which has to modify some paths and query parameters. When doing something like below, the `subject` query parameter is not set at the end.
```java
public class ContextFilter implements ContainerRequestFilter {
    public void filter(ContainerRequestContext requestContext) throws IOException {
                UriBuilder builder = requestContext.getUriInfo().getRequestUriBuilder();
                builder.replacePath("/newpath");
                builder.replaceQueryParam("subject", "test");
                uri = builder.build();
                requestContext.setRequestUri(uri);

```

Is there a workaround or a better way doing this?

**Comment by geoand** [comment]
Interesting. There doesn't seem to be workaround currently.

Do you mind attaching a sample project that we can test any potential fixes against? This will help us speed up the process of providing a fix that we can backport into some `2.13.z`  release.

Thanks

**Comment by spike83** [comment]
Here is a minimal reproducer.
[quarkus-28555.zip](https://github.com/quarkusio/quarkus/files/9782821/quarkus-28555.zip)
